cpp/dolfinx/graph/Partitioning.h

```diff
@@ -118,11 +118,11 @@ class Partitioning
   ///   row). The global index for the [0, ..., n) local rows is assumed
   ///   to be the local index plus the offset for this process
   /// @return The data for each index in @p indices
-  static Eigen::Array<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>
-  distribute_data(
-      MPI_Comm comm, const std::vector<std::int64_t>& indices,
-      const Eigen::Ref<const Eigen::Array<double, Eigen::Dynamic,
-                                          Eigen::Dynamic, Eigen::RowMajor>>& x);
+  template <typename T>
```

Comment:
What would happen if one user will try to use this for T different from int, int64_t or double? A linker error?

Reply:
Yes. I do it the other way - make implementation a template and overload for user interface.

Reply:
Since it won't be usable it shouldn't be exposed. See comment above on moving implementation into an anonymous namespace.